### DISCUSSION TITLE
Implement a Planning mode

### DISCUSSION THREAD

**Discussion by fin0** [iSoron/uhabits#289] [Feature Requests]
Hey,
I have a suggestion that is in the back of my head for a rather long time.
What about having something like a planning mode in the app? Implemented as a checkbox like the "show archived" option. If selected, the days-timeline of each habit, where you normally select a habit as "done" for the past days, will also expand to the future. Now, maybe with a new UI-gesture (like fast double clicking), one could make the selection of a habit in the future possible. The selection has no implication, just visually shows what has to be done at that day. Using another symbol or another color for the tick-mark would be advisable. If that selection is not actively marked as done in the old style if the future day becomes today, it is just deleted at midnight. If a new gesture is implemented, one could also allow this to be done on today (even without planning mode switched on). In the morning one could mark things to be done today.
Hope this is understandable :) If not, just ask, I will clarify.

Have a nice day you all.

**Comment by iSoron** [comment]
Thank you for the suggestion, @fin0. If I understand correctly, the main use case for this feature would be marking things that should be done today, so that, even if you have many habits, it's still easy to tell which ones should you be working on. Is this correct? Do you have another use case in mind?

If this is the only use case, then I think a planning mode is too complicated for this. My guess is that most users would be surprised when the checkmarks they added in the future disappear.

We could, instead, allow the user to swipe habits away or snooze them on the main screen. This would make the habit disappear until the next day. The user could open the app every day in the morning and go through the list, swiping away things they don't want to do today.

**Comment by fin0** [comment]
Thanks for the reply. Yes you understood correctly, main use case would be to mark habits that should be done. Today would be most important, but I would also like to have this feature available for the next days (optionally expanding the timeline to the future)

You are right, if tickmarks would disappear, that would be bad for the useability. But I'm also not completely convinced by the idea of swiping habits away for today, because one could not play around with different configurations of which habits should be done for today. Sometimes maybe you just want to look at the marked habits and decide that this selection is too less for today and exchange some of them.
To solve this problem of disappearing tickmarks: What if we just use another symbol instead of the cross together with a new UI-gesture. For example you could double click on a habit (in the today column) and then this habit would be marked as planned, maybe by using the symbol "P" instead of the tickmark. Double-clicking again would delete this. Marking the habit in the normal way would just delete the "P" if there is one and replacing it with the tick.
What do you think about this?
